src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -548,7 +547,29 @@ internal List<INodeLibraryDependencyInfo> NodeLibraryDependencies
                             packageDependencies[saved] = new PackageDependencyInfo(saved);
                         }
                         packageDependencies[saved].AddDependent(node.GUID);
-                        packageDependencies[saved].IsLoaded = saved.Equals(collected);
+
+                        if (collected != null)
+                        {
+                            if (saved.Name == collected.Name)
```

Comment:
I think it might be helpful to label these cases

Reply:
@mjkkirschner @scottmitchell .I have added the fix like we discussed.